stdlib/array.ml

```diff
@@ -292,3 +292,27 @@ let stable_sort cmp a =
 
 
 let fast_sort = stable_sort
+
+(** {6 Iterators} *)
+
+let to_iter a =
+  let next i =
+    if i < length a
```

Comment:
Should `length a` be pulled above the definition of `next` to avoid the check against every element?